clap_complete/src/shells/fish.rs

```diff
@@ -67,22 +67,27 @@ fn gen_fish_inner(
             basic_template.push_str(" -n \"__fish_use_subcommand\"");
         }
     } else {
-        basic_template.push_str(
-            format!(
-                " -n \"{}\"",
-                parent_commands
-                    .iter()
-                    .map(|command| format!("__fish_seen_subcommand_from {command}"))
-                    .chain(
-                        cmd.get_subcommands()
-                            .flat_map(Command::get_name_and_visible_aliases)
-                            .map(|name| format!("not __fish_seen_subcommand_from {name}"))
-                    )
-                    .collect::<Vec<_>>()
-                    .join("; and ")
-            )
-            .as_str(),
-        );
+        // do not generate completion when "help" is an auto-generated subcommand.
+        if matches!(parent_commands, ["help"]) && cmd.is_disable_help_subcommand_set() {
```

Comment:
I don't really understand  `is_disable_help_subcommand_set`, I fail trying
```suggestion
        if matches!(parent_commands, ["help"]) && !cmd.is_disable_help_subcommand_set() {
```